datastream-server/src/main/java/com/linkedin/datastream/server/assignment/BroadcastStrategy.java

```diff
@@ -59,17 +60,27 @@ public BroadcastStrategy(Optional<Integer> maxTasks) {
 
     Map<String, Set<DatastreamTask>> newAssignment = new HashMap<>();
 
+    Set<DatastreamTask> tasksAvailableToReuse = new HashSet<>();
     // Make a copy of the current assignment, since the strategy modifies it during calculation
     Map<String, Set<DatastreamTask>> currentAssignmentCopy = new HashMap<>(currentAssignment.size());
-    currentAssignment.forEach((k, v) -> currentAssignmentCopy.put(k, new HashSet<>(v)));
+    currentAssignment.forEach((k, v) -> {
+      currentAssignmentCopy.put(k, new HashSet<>(v));
+      // building the full list and then removing the reused tasks rather than building the list from dead tasks
+      // to avoid assignment of same task from dead instance to two live instances in case of interrupt exception.
```

Comment:
"removing the reused tasks"? Did you mean removing the tasks assigned to current live instances? 

> to avoid assignment of same task from dead instance to two live instances in case of interrupt exception.

So that part of the comment is a bit confusing here. What got interrupted? This will only make sense to people who already know the details about the problem you are solving. Instead can you modify this to talk about interrupted cleanup of ZK nodes due to shutdown during leader assignment?

Reply:
Strongly agree.

There's a whole story behind reusing tasks assigned to dead instances. I'd suggest expanding the comment even further to explain:
- why we reuse tasks assigned to dead instances to begin with
- caveats we consider during this reuse, like the possibility of having the same task assigned to more than one instance (and why that might happen to begin with)

Reply:
Done.